### PR TITLE
Update for LIST ENGINES

### PR DESCRIPTION
**Pull request by AlchemistCH**
My proposition for #1251

### CODE REVIEW

**Comment by Dunbaratu** [comment]
@hvacengi, Is this a candidate for 0.19.0, since it's in testing phase?

**Comment by hvacengi** [comment]
Yeah, it's a candidate.  I've been thinking about the implications of it.  Obviously the list needs to be updated to properly handle the `MultiModeEngine`.  But I don't know why it previously only listed active engines, so I don't really have the other half of the argument to weigh.  

I've honestly never really understood why the two methods (`list [parts] in [list var]` and `list [parts]` didn't actually use the same method, so I'd be tempted to modify it so that it uses the same part factory method.  That would mean only one location in the code to maintain the list generation, which would prevent it from getting out of sync in the future.

**Comment by AlchemistCH** [comment]
Yeah, maybe calling the same list generation and then just displaying the list by these functions would be better.
Ah, the amount of frustration trying to to fix one function while unknowingly calling another one for tests. And then finally finding this mess.

**Comment by hvacengi** [comment]
Does that mean you can update it to call that method instead, or should I make the edit and submit a PR to you @AlchemistCH?

**Comment by AlchemistCH** [comment]
OK, I'll deal with this file.

**Comment by AlchemistCH** [comment]
Here, made LIST ENGINES function to use the one from EngineValue. 
(Is it OK to take contents of ListValue like this? Seems perfectly functional)

Was also considering doing the same to LIST SENSORS, but that would require changing some access levels in SensorValue. Besides that's way simpler function.

File: src/kOS/Function/PrintList.cs [hunk 1]
```diff
@@ -276,7 +273,7 @@ private kList GetConfigList()
             list.AddColumn("Name", 34, ColumnAlignment.Left);
             list.AddColumn("Value", 6, ColumnAlignment.Left);
 
-            foreach (ConfigKey key in Config.Instance.GetConfigKeys())
+            foreach (ConfigKey key in SafeHouse.Config.GetConfigKeys())
```
Comment (hvacengi) [inline src/kOS/Function/PrintList.cs]: The interface exposed by `SafeHouse.Config` does not include `GetConfigKeys()`.  I'm going to revert this back to `Config.Instance` in the merge.

Comment (AlchemistCH) [inline src/kOS/Function/PrintList.cs]: Ok, I've put it back. Shouldn't have copied over the entire file when the difference on my side was a forgotten 'uses' reference.

**Comment by hvacengi** [comment]
I just wanted to share an example of my testing of the PR:
screenshot55

**Comment by hvacengi** [comment]
Apparently I fail at merging.  I merged this into my develop branch and pushed, and the commits are now shown in the history of develop... but the merge did not show up here.

I'll try to figure that out.

**Pull request merged**